crates/bevy_ecs/src/query/access.rs

```diff
@@ -325,6 +354,57 @@ impl<T: SparseSetIndex> FilteredAccess<T> {
     }
 }
 
+// A struct to express something like `Or<(With<A>, With<B>)>`.
+// Filters like `(With<A>, Or<(With<B>, With<C>)>` are expanded into `Or<(With<(A, B)>, With<(B, C)>)>`.
```

Comment:
The proper syntax for `and`ed `With`es is a tuple around the `With` and not inside. (Also fixes the wrong `With<(B, C)`)
```suggestion
// Filters like `(With<A>, Or<(With<B>, With<C>)>` are expanded into `Or<((With<A>, With<B>), (With<A>, With<C>))>`.
```

Reply:
Ah, good catch, thanks